### PR TITLE
Improve documentation RequestHeaders and ResponseHeaders

### PR DESCRIPTION
**Pull request by koenekelschot**
Summary of the changes (Less than 80 chars)
 - Added xmldocs

Addresses #3465

### CODE REVIEW

**Comment by jkotalik** [comment]
@anurse your favorite kind of PR 😄.

File: src/Http/Http.Extensions/src/RequestHeaders.cs [hunk 1]
```diff
@@ -279,11 +279,26 @@ public Uri Referer
             }
         }
 
+        /// <summary>
+        /// Gets the value of header with <paramref name="name"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the header.
+        /// The given type must have a static TryParse method</typeparam>
+        /// <param name="name">The name of the header to retrieve.</param>
+        /// <returns>The value of the header</returns>
```
Comment (jkotalik) [inline src/Http/Http.Extensions/src/RequestHeaders.cs]: As a follow-up, we should either make another PR or add to this PR to add xml documentation for the rest of these properties. 

Also, isn't there an analyzer that we can enable to check for all public apis that don't have xml docs?

Comment (anurse) [reply src/Http/Http.Extensions/src/RequestHeaders.cs]: One small nit:

```suggestion
        /// <returns>The value of the header.</returns>
```

> Also, isn't there an analyzer that we can enable to check for all public apis that don't have xml docs?

It's even better than that :). If we turn the right settings on we get a build warning (or error if we promote it to that). Some projects are using that now. We need to get "clean" before we can turn that on more broadly though.

File: src/Http/Http.Extensions/src/RequestHeaders.cs [hunk 2]
```diff
@@ -279,11 +279,26 @@ public Uri Referer
...
         }
 
+        /// <summary>
+        /// Gets the values of header with <paramref name="name"/>.
+        /// </summary>
+        /// <remarks><typeparamref name="T"/> must contain a TryParseList method with the signature <code>public static bool TryParseList(IList&lt;string&gt;, out IList&lt;T&gt;)</code></remarks>
+        /// <typeparam name="T">The type of the header.
+        /// The given type must have a static TryParseList method</typeparam>
+        /// <param name="name">The name of the header to retrieve.</param>
+        /// <returns>List of values of the header</returns>
```
Comment (anurse) [inline src/Http/Http.Extensions/src/RequestHeaders.cs]:
```suggestion
        /// <returns>List of values of the header.</returns>
```

File: src/Http/Http.Extensions/src/RequestHeaders.cs [hunk 3]
```diff
@@ -279,11 +279,26 @@ public Uri Referer
...
+        /// <returns>The value of the header</returns>
         public T Get<T>(string name)
         {
             return Headers.Get<T>(name);
         }
 
+        /// <summary>
+        /// Gets the values of header with <paramref name="name"/>.
+        /// </summary>
+        /// <remarks><typeparamref name="T"/> must contain a TryParseList method with the signature <code>public static bool TryParseList(IList&lt;string&gt;, out IList&lt;T&gt;)</code></remarks>
```
Comment (anurse) [inline src/Http/Http.Extensions/src/RequestHeaders.cs]: You used `typeparamref` here but not in the `Get` method. I'm fine with doing this either way but we should be consistent.

Similarly, you put the full signature in here, but not above.

Comment (anurse) [reply src/Http/Http.Extensions/src/RequestHeaders.cs]: We should also be consistent between `remarks` and `typeparam` items. Let's make them similar in format and consistent between both methods.

File: src/Http/Http.Extensions/src/ResponseHeaders.cs [hunk 4]
```diff
@@ -158,11 +158,27 @@ public IList<SetCookieHeaderValue> SetCookie
...
+        /// <returns>The value of the header</returns>
         public T Get<T>(string name)
         {
             return Headers.Get<T>(name);
         }
 
+        /// <summary>
+        /// Gets the values of header with <paramref name="name"/>.
+        /// </summary>
+        /// <remarks><typeparamref name="T"/> must contain a TryParseList method with the signature <code>public static bool TryParseList(IList&lt;string&gt;, out IList&lt;T&gt;)</code></remarks>
```
Comment (anurse) [inline src/Http/Http.Extensions/src/ResponseHeaders.cs]: Same comments about consistency apply here.

**Review by analogrelay — CHANGES_REQUESTED** [review]
Generally looks good! Let's make all the methods consistent in how the docs are formatted. Once we've got that, I think we'll be good to go.

File: src/Http/Http.Extensions/src/RequestHeaders.cs (hunk 1 above)
Comment (peckham) [reply src/Http/Http.Extensions/src/RequestHeaders.cs]: @jkotalik and @anurse thanks for the comments! They should be resolved in the latest commit.

Regarding the other properties in these classes, I'm willing to document them as well, but do add those docs to this PR or do I open a new PR?

**Comment by Tratcher** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=387600

**Review by Tratcher — APPROVED** [review]

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/3241
https://github.com/aspnet/AspNetCore-Internal/issues/3128
https://github.com/aspnet/AspNetCore-Internal/issues/3173

**Review by analogrelay — APPROVED** [review]

**Comment by Tratcher** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=387600

**Comment by Tratcher** [comment]
@HaoK Build Tests: Helix x64 still looks too flaky to be required.

**Comment by HaoK** [comment]
Looks like there might be something going on with redis causing these tests to fail:

```
   System.AggregateException : One or more errors occurred. (Command '/usr/bin/docker run --rm -p 6379:6379 --name redisTestContainer -d redis' failed with exit code '125'. Output:
Unable to find image 'redis:latest' locally
/usr/bin/docker: Error response from daemon: received unexpected HTTP status: 503 Service Unavailable.
See '/usr/bin/docker run --help'.
```

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**